includes/admin/class-wc-rest-payments-payment-intents-controller.php

```diff
@@ -46,4 +74,47 @@ public function get_payment_intent( $request ) {
 
 		return $this->forward_request( 'get_intent', [ $payment_intent_id ] );
 	}
+
+	/**
+	 * Create a payment intent.
+	 *
+	 * @param WP_REST_Request $request data about the request.
+	 */
+	public function post_payment_intent( $request ) {
+		try {
+
+			$order_id = $request->get_param( 'order_id' );
+			$order    = wc_get_order( $order_id );
+			$currency = strtolower( $order->get_currency() );
+			$amount   = WC_Payments_Utils::prepare_amount( $order->get_total(), $currency );
+			$name     = sanitize_text_field( $order->get_billing_first_name() ) . ' ' . sanitize_text_field( $order->get_billing_last_name() );
+			$email    = sanitize_email( $order->get_billing_email() );
+			$metadata = [
+				'customer_name'  => $name,
+				'customer_email' => $email,
+				'site_url'       => esc_url( get_site_url() ),
+				'order_id'       => $order->get_id(),
+				'order_number'   => $order->get_order_number(),
+				'order_key'      => $order->get_order_key(),
+				'payment_type'   => 'single',
+			];
+
+			$wcpay_server_request = Create_And_Confirm_Intention::create();
```

Comment:
Have you considered using `SetupIntents` instead of 'regular' intents, since we might charge this intent later on? I am not saying that this is wrong, I am just curious would `SetupIntents` be a better option here.

Reply:
Discussed here - pcreKM-20q-p2#comment-3005